File: 프로그래머스연습문제/짝지어 제거하기.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int solution(string s)
{
    int answer = 0;
    
    vector <char> st;
 
    for(int i = 0; i < s.size(); i++){
        if(st.empty()){
            st.push_back(s[i]);
        }
        else{
            if(st.back() == s[i]){
                st.pop_back();

            }
            else{
                st.push_back(s[i]);
            }
        }
    }
    
  
    if(st.empty())answer = 1;
    else answer = 0;
    

    return answer;
}

File: 프로그래머스_python/연습문제/level2/짝지어 제거하기.py
def solution(s):
    answer = 0
    
    if len(s) % 2 != 0:
        return 0
    
    array = list(s)
    stack = []
    count = 0
    
    for i in array:
        
        if len(stack) == 0:
            stack.append(i)
            
        elif i == stack[-1]:
            stack.pop()
            count += 1
            
        else:
            stack.append(i)
        
        
    if count == len(array)//2:
        answer = 1
        
    

    return answer
